### PR TITLE
Dependent Fields: Show dependent fields in Merge expenses page

### PR DESCRIPTION
**Pull request by yksolanki9**

### CODE REVIEW

File: src/app/fyle/merge-expense/merge-expense.page.ts [hunk 1]
```diff
@@ -270,7 +274,6 @@ export class MergeExpensePage implements OnInit {
     this.loadCustomFields$ = new BehaviorSubject(this.genericFieldsForm.value?.category);
 
     this.setupCustomInputs();
-    this.generateCustomInputOptions();
```
Comment (yksolanki9) [inline src/app/fyle/merge-expense/merge-expense.page.ts:273]: Duplicate call here - not needed, ref line 282 in new changes

File: src/app/core/services/merge-expenses.service.ts [hunk 2]
```diff
@@ -588,6 +592,33 @@ export class MergeExpensesService {
       .filter((element) => element !== undefined);
   }
 
+  getProjectDependentFieldsMapping(
```
Comment (Julias0) [inline src/app/core/services/merge-expenses.service.ts:595]: Lets add a unit test just for this method

File: src/app/fyle/merge-expense/merge-expense.page.ts [hunk 3]
```diff
@@ -587,6 +600,20 @@ export class MergeExpensePage implements OnInit {
...
+      switchMap((fields) => {
+        const customFields = this.customFieldsService.standardizeCustomFields([], fields);
+        return of(customFields);
+      })
+    );
+
+    this.projectDependentFieldsMapping$ = dependentFields$.pipe(
+      map((dependentFields) =>
+        this.mergeExpensesService.getProjectDependentFieldsMapping(this.expenses, dependentFields)
+      )
```
Comment (Julias0) [inline src/app/fyle/merge-expense/merge-expense.page.ts:615]: add a shareReplay(1), to avoid a 2nd api call when `mergeExpense` method is called

Comment (yksolanki9) [reply src/app/fyle/merge-expense/merge-expense.page.ts:615]: Done

File: src/app/core/services/merge-expenses.service.ts (hunk 2 above)
Comment (yksolanki9) [reply src/app/core/services/merge-expenses.service.ts:595]: Added

Comment (yksolanki9) [reply src/app/core/services/merge-expenses.service.ts:595]: Moved to a separate PR as per our discussion - https://github.com/fylein/fyle-mobile-app/pull/1668

**Review by Julias0 — APPROVED** [review]

**Pull request merged**